### PR TITLE
calling openwhisk from action fails because api host is set to "localhost" #58

### PR DESCRIPTION
**Pull request by alexkli**
For #58:
* using `hostname` to detect the host name of the Docker host instead of falling back to `localhost`
* removing unnecessary `https://` prefix on `WHISK_API_HOST_NAME`

### CODE REVIEW

**Comment by alexkli** [comment]
Travis build failed with
```
waiting for the Whisk controller to come up ... 
...
The job exceeded the maximum time limit for jobs, and has been terminated.
```
Maybe the Travis environment returns no useful `hostname`, even failing the basic OW docker-compose setup?

**Comment by alexkli** [comment]
[Travis build](https://travis-ci.org/apache/incubator-openwhisk-devtools/builds/282493559) failed with

```
waiting for the Whisk controller to come up ... 
...
The job exceeded the maximum time limit for jobs, and has been terminated.
```

Maybe the Travis environment returns no useful `hostname`, even failing the basic OW docker-compose setup?

**Comment by alexkli** [comment]
Well, the [travis build](https://travis-ci.org/apache/incubator-openwhisk-devtools/builds/282496097) for #57 failed with the same error, and that one is just a readme change.

**Comment by ddragosd** [comment]
There's an error with the latest changes in the upstream that I'm working to fix on master and then it should unblock this build too.

**Comment by ddragosd** [comment]
@alexkli feel free to merge with `master` branch to fix the build issues.

File: docker-compose/Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-DOCKER_HOST_IP ?= $(shell echo ${DOCKER_HOST} | grep -o "[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}" || echo localhost)
+DOCKER_HOST_IP ?= $(shell echo ${DOCKER_HOST} | grep -o "[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}" || hostname)
```
Comment (ddragosd) [inline docker-compose/Makefile]: 👍  `hostname` is better than localhost.

In some cases it won't be resolved from within the container. Should we document that the developer needs to ensure the hostname is resolvable ?

Comment (alexkli) [reply docker-compose/Makefile]: Yes.

**Comment by alexkli** [comment]
Ok, I figured using `hostname` is not a good idea. See #58. Will come up with a new patch. Closing this one.

**Pull request closed without merging**